packages/algorand-js/src/runtime/runtime.ts

```diff
@@ -0,0 +1,215 @@
+import { TealError } from "../errors/errors";
+import { ERRORS } from "../errors/errors-list";
+import { Interpreter } from "../index";
+import { BIGINT0 } from "../interpreter/opcode-list";
+import { checkIndexBound, compareArray } from "../lib/compare";
+import { convertToString } from "../lib/parsing";
+import { assertValidSchema, getKeyValPair } from "../lib/stateful";
+import type { Context, SDKAccount, StackElem, State, Txn } from "../types";
+
+export class Runtime {
+  interpreter: Interpreter;
+  store: State;
+
+  constructor (interpreter: Interpreter) {
+    this.interpreter = interpreter;
+    this.store = {
+      accounts: new Map<string, SDKAccount>(),
+      globalApps: new Map<number, SSCParams>()
+    };
+  }
+
+  assertAccountDefined (a?: SDKAccount): SDKAccount {
+    if (a === undefined) {
+      throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
+    }
+    return a;
+  }
+
+  assertAppDefined (appId: number): SSCParams {
+    const globalDelta = this.interpreter.ctx.state.globalApps.get(appId);
+    if (globalDelta === undefined) {
+      throw new TealError(ERRORS.TEAL.APP_NOT_FOUND);
+    }
+    return globalDelta;
+  }
+
+  getAccount (accountIndex: bigint): SDKAccount {
+    let account: SDKAccount | undefined;
+    if (accountIndex === BIGINT0) {
+      const senderAccount = convertToString(this.interpreter.ctx.tx.snd);
+      account = this.interpreter.ctx.state.accounts.get(senderAccount);
+    } else {
+      accountIndex--;
+      checkIndexBound(Number(accountIndex), this.interpreter.ctx.tx.apat);
+      const pkBuffer = this.interpreter.ctx.tx.apat[Number(accountIndex)];
+      account = this.interpreter.ctx.state.accounts.get(convertToString(pkBuffer));
+    }
+    return this.assertAccountDefined(account);
+  }
+
+  /**
+   * Description: fetches global state value for key present app's global data
+   * returns undefined otherwise
+   * @param appId: current application id
+   * @param key: key to fetch value of from local state
+   */
+  getGlobalState (appId: number, key: Uint8Array): StackElem | undefined {
+    const appDelta = this.assertAppDefined(appId);
+    const globalState = appDelta["global-state"];
+
+    const keyValue = globalState.find(schema => compareArray(schema.key, key));
+    const value = keyValue?.value;
+    if (value) {
+      return value?.bytes || BigInt(value?.uint);
+    }
+    return undefined;
+  }
+
+  /**
+   * Description: add new key-value pair or updating pair with existing key in
+   * app's global data for application id: appId, throw error otherwise
+   * @param appId: current application id
+   * @param key: key to fetch value of from local state
+   * @param value: key to fetch value of from local state
+   */
+  updateGlobalState (appId: number, key: Uint8Array, value: StackElem): SSCStateSchema[] {
+    const appDelta = this.assertAppDefined(appId);
+    const globalState = appDelta["global-state"];
```

Comment:
as above, let's rename to: app, and appGlobalState.

Reply:
done